File: services/notification-service/src/main/java/de/denizaltun/notificationservice/model/Alert.java
package de.denizaltun.notificationservice.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "alerts", indexes = {
        @Index(name = "idx_alert_vehicle_id", columnList = "vehicle_id"),
        @Index(name = "idx_alert_type", columnList = "alert_type"),
        @Index(name = "idx_alert_status", columnList = "status")
})
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Alert {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "vehicle_id", nullable = false)
    private String vehicleId;

    @Enumerated(EnumType.STRING)
    @Column(name = "vehicle_type")
    private VehicleType vehicleType;

    @Enumerated(EnumType.STRING)
    @Column(name = "alert_type", nullable = false)
    private AlertType alertType;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private AlertStatus status;

    @Column(nullable = false)
    private String message;

    @Column(name = "threshold_value")
    private Double thresholdValue;

    @Column(name = "actual_value")
    private Double actualValue;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "acknowledged_at")
    private LocalDateTime acknowledgedAt;

    @Column(name = "resolved_at")
    private LocalDateTime resolvedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        if (status == null) {
            status = AlertStatus.ACTIVE;
        }
    }
}

File: frontend/src/types/alert.ts
import type { VehicleType } from "./vehicle";

export type AlertType =
    | "LOW_FUEL"
    | "HIGH_ENGINE_TEMP"
    | "LOW_BATTERY"
    | "VEHICLE_IDLE_TOO_LONG"
    | "EMERGENCY_STATUS_CHANGE";

export type AlertStatus = "ACTIVE" | "ACKNOWLEDGED" | "RESOLVED";

export interface Alert {
    id: number;
    vehicleId: string;
    vehicleType: VehicleType;
    alertType: AlertType;
    status: AlertStatus;
    message: string;
    thresholdValue: number;
    actualValue: number;
    createdAt: string;
    acknowledgedAt?: string;
    resolvedAt?: string;
}